### PR TITLE
extract Graveyard from Player

### PR DESCRIPTION
**Pull request by chrisvaughn**
* Extract graveyard data into it's own singleton
* migration for data

### CODE REVIEW

**Review by rj-lee — DISMISSED** [review]
one very small change and then LGTM

File: myning/objects/graveyard.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+    @classmethod
+    @property
+    def file_name(cls):
+        return "graveyard"
+
+    def __init__(
+        self,
+        soul_credits: float = 0,
+        fallen_allies: List[Character] = None,
```
[the remark this answers is not in the document]
Comment (TheRedPanda17) [reply myning/objects/graveyard.py:25]: Agreed

File: main.py [hunk 2]
```diff
@@ -28,15 +29,17 @@ def main():
...
     ResearchFacility.initialize()
     Settings.initialize()
     Stats.initialize()
     Trip.initialize()
 
-    # This ensures new players have the new migrations. Preferably, we'd loop through the
-    # MIGRATIONS, but we have a circular dependency if we do, so this is the hack right now.
-    Player().completed_migrations = [1, 2, 3, 4, 5, 6, 7, 8]
+    # import MIGRATIONS here to resolve circular dependencies
+    from myning.migrations import MIGRATIONS  # pylint: disable=import-outside-toplevel
```
Comment (TheRedPanda17) [inline main.py:40]: Nice! This has been a pain point until now. I always forget you can import at other places

**Review by TheRedPanda17 — APPROVED** [review]
I'd say just fix the typing.List to list and it's good. Thanks for simplifying the player object!

File: myning/objects/graveyard.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (chrisvaughn) [reply myning/objects/graveyard.py]: updated pushed.  I guess I always assumed if I wanted to specify the type of the list I had to use typing.List.  this is great to know.

**Review by rj-lee — APPROVED** [review]

File: myning/objects/graveyard.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rj-lee) [reply myning/objects/graveyard.py]: this used to be the case but some newer python version allows us to use `list`, and it's nice

**Review by TheRedPanda17 — APPROVED** [review]

**Pull request merged**